File: FizzBuzz/FizzBuzz.cs
using System;

namespace FizzBuzz
{
    public class FizzBuzz
    {
        private readonly int FIZZ = 3;
        private readonly int BUZZ = 5;

        private bool isFizz(int num) 
        {
            return num % FIZZ == 0;
        }

        private bool isBuzz(int num) 
        {
            return num % BUZZ == 0;
        }

        public void fizzbuzz(int num = 100)
        {
            for (int i = 1; i <= num; i++)
            {
                if (isFizz(i) && isBuzz(i))
                {
                    Console.Write("FizzBuzz");
                }
                else if (isFizz(i))
                {
                    Console.Write("Fizz");
                }
                else if (isBuzz(i))
                {
                    Console.Write("Buzz");
                }
                else
                {
                    Console.Write(i);
                }
                Console.Write("\n");
            }
        }
    }

    public class FizzBuzzImpl
    {
        static void Main()
        {
            var fz = new FizzBuzz();
            fz.fizzbuzz();
        }
    }
}

File: FizzBuzz/Fizzbuzz.ts
class FizzBuzz {
    private readonly FIZZ:number = 3;
    private readonly BUZZ:number = 5;

    private isFizz(num:number) : boolean {
        return num % this.FIZZ === 0;
    }
    private isBuzz(num:number) : boolean {
        return num % this.BUZZ === 0;
    }

    public fizzbuzz(num:number = 100) : void {
        for (let i = 1; i <= num; i++) {
            if (this.isFizz(i) && this.isBuzz(i)) {
                console.log("FizzBuzz");
            } else if (this.isFizz(i)) {
                console.log("Fizz");
            } else if (this.isBuzz(i)) {
                console.log("Buzz");
            } else {
                console.log(i);
            }
        }
    }
}

const fz = new FizzBuzz();
fz.fizzbuzz();